fargate/check.py

```diff
@@ -0,0 +1,158 @@
+IO_METRICS = {'io_service_bytes_recursive':'fargate.io.bytes.', 'io_serviced_recursive':'fargate.io.ops.'}
+
+class FargateCheck(AgentCheck):
+
+    def check(self, instance):
+        timeout = float(instance.get('timeout', DEFAULT_TIMEOUT))
+        metadata_endpoint = API_ENDPOINT + METADATA_ROUTE
+        stats_endpoint = API_ENDPOINT + STATS_ROUTE
+
+        try:
+            request = requests.get(metadata_endpoint, timeout=timeout)
+        except requests.exceptions.Timeout as e:
+            msg = 'Fargate {} endpoint timed out after {} seconds'.format(metadata_endpoint, timeout)
+            self.service_check('fargate_check', AgentCheck.CRITICAL, message=msg)
+            self.log.error(msg)
+            return
+        except Exception as e:
+            msg = 'Error fetching Fargate {} endpoint'.format(metadata_endpoint)
+            self.service_check('fargate_check', AgentCheck.CRITICAL, message=msg)
+            self.log.error(msg)
+            return
+
+        if request.status_code != 200:
+            msg = 'Fargate {} endpoint responded with {} HTTP code'.format(metadata_endpoint, request.status_code)
+            self.service_check('fargate_check', AgentCheck.CRITICAL, message=msg)
+            self.log.warning(msg)
+            return
+
+        metadata = {}
+        try:
+            metadata = request.json()
+        except:
+            msg = 'Cannot decode Fargate {} endpoint response'.format(metadata_endpoint)
+            self.service_check('fargate_check', AgentCheck.WARNING, message=msg)
+            self.log.warning(msg)
+            return
+
+        if not all (k in metadata for k in ("Cluster","Containers")):
+            msg = 'Missing critical metadata in {} endpoint response'.format(metadata_endpoint)
+            self.service_check('fargate_check', AgentCheck.WARNING, message=msg)
+            self.log.warning(msg)
+            return
+
+        common_tags = instance.get('tags', [])
+        common_tags.append('ecs_cluster:' + metadata['Cluster'])
+        common_tags.append('ecs_task_family:' + metadata['Family'])
+        common_tags.append('ecs_task_version:' + metadata['Version'])
+        label_whitelist = instance.get('label_whitelist', [])
+
+        container_tags = {}
+        for container in metadata['Containers']:
+            c_id = container['DockerId']
+            container_tags[c_id] = []
+            container_tags[c_id].extend(common_tags)
+            container_tags[c_id].append('docker_image:' + container['Image'])
+            image_split = container['Image'].split(':')
+            container_tags[c_id].append('image_name:' + ':'.join(image_split[:-1]))
+            container_tags[c_id].append('image_tag:' + image_split[-1])
+            for label, value in container["Labels"].iteritems():
+                if label in label_whitelist or label not in LABEL_BLACKLIST:
+                    container_tags[c_id].append(label + ':' + value)
+
+            if container['Limits']['CPU'] > 0:
```

Comment:
any reason we're not getting the memory limit from here?

Reply:
@hkaj you can but you'll not get the metric for the ecs pause container, getting it at the container level `container_stats['memory_stats']['limit']` allows it

Reply:
ok makes sense 👍